src/test/java/org/openstreetmap/atlas/geography/atlas/items/RelationFlatteningRule.java

```diff
@@ -0,0 +1,41 @@
+package org.openstreetmap.atlas.geography.atlas.items;
+
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.utilities.testing.CoreTestRule;
+import org.openstreetmap.atlas.utilities.testing.TestAtlas;
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Node;
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Relation;
+import org.openstreetmap.atlas.utilities.testing.TestAtlas.Relation.Member;
+
+/**
+ * @author samuelgass
+ */
+public class RelationFlatteningRule extends CoreTestRule
+{
+
+    @TestAtlas(
+
+            nodes = { @Node(id = "1"), @Node(id = "2"), @Node(id = "3"), @Node(id = "4"),
+                    @Node(id = "5") },
+
+            relations = {
+
+                    @Relation(id = "6", members = {
+                            @Member(id = "1", role = "outside", type = "node") }),
+                    @Relation(id = "7", members = {
+                            @Member(id = "1", role = "outside", type = "node"),
+                            @Member(id = "2", role = "outside", type = "node") }),
+                    @Relation(id = "8", members = {
```

Comment:
How about testing a valid relation with one or more sub-relations? Currently this is only testing an invalid relationship. It may be good to separate these into individual tests and test atlases. Something like below. Thoughts?

- testShallowRelation
- testNestedRelation
- testLoopingRelation

Reply:
good idea